### ISSUE TITLE
[java] JUnitAssertionsShouldIncludeMessage false positive with method call

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#2883]
**Affects PMD Version:** 6.30.0-SNAPSHOT

**Rule:** [JUnitAssertionsShouldIncludeMessage](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#junitassertionsshouldincludemessage)

**Description:**

From https://github.com/pmd/pmd/pull/2871#issuecomment-717848659

**Code Sample demonstrating the issue:**

https://chunk.io/pmd/fc7db65b2c6a46eca4c9a0c3012482e2/diff/spring-framework/index.html#A6120

```java
import static org.junit.Assert.assertEquals;

public class AssertionMessageTest {
    public void bar(int expectedLength) {
        try {
            Integer.parseInt("a");
        } catch (RuntimeException e) {
            assertEquals(e.toString(), expectedLength, e.getMessage().length()); // no warning expected
        }
    }
}
```

**Expected outcome:**

*   Does PMD report a violation, where there shouldn't be one? -> false-positive

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

**Comment by oowekyala** [comment]
I'm pretty sure this is fixed in #2899

**Comment by adangel** [comment]
Fixed via #2899 for PMD 7.

**Comment by adangel** [comment]
Fixed via #2899 for PMD 7.
